### ISSUE TITLE
Issue with pre-commit (window 10)

### ISSUE THREAD

**Issue by idalr** [ArneBinder/pie-document-level#119]
Currently, all my pre-commits fail when I commit-and-push or run `pre-commit run -a` in Terminal. 
The error it throws is...
```
[INFO] Installing environment for https://github.com/pre-commit/pre-commit-hooks.
[INFO] Once installed this environment will be reused.
[INFO] This may take a few minutes...
An unexpected error has occurred: CalledProcessError: command: ('C:\\Users\\Rinrada\\.cache\\pre-commit\\repoyj49m085\\py_env-default\\Scripts\\python.EXE', '-mpip', 'install', '.')
return code: 1
stdout:
    Processing c:\users\rinrada\.cache\pre-commit\repoyj49m085
      Preparing metadata (setup.py): started
      Preparing metadata (setup.py): finished with status 'done'
    Could not fetch URL https://pypi.org/simple/ruamel-yaml/: There was a problem confirming the ssl certificate: HTTPSConnectionPool(host='pypi.org', port=443): Max retries exceeded with url: /simple/ruamel-yaml/ (Caused by SSLError("Can't connect to HTTPS URL because the SSL module is not available.")) - skipping
    INFO: pip is looking at multiple versions of pre-commit-hooks to determine which version is compatible with other requirements. This could take a while.
stderr:
    WARNING: pip is configured with locations that require TLS/SSL, however the ssl module in Python is not available.
    WARNING: Retrying (Retry(total=4, connect=None, read=None, redirect=None, status=None)) after connection broken by 'SSLError("Can't connect to HTTPS URL because the SSL module is not available.")': /simple/ruamel-yaml/
    WARNING: Retrying (Retry(total=3, connect=None, read=None, redirect=None, status=None)) after connection broken by 'SSLError("Can't connect to HTTPS URL because the SSL module is not available.")': /simple/ruamel-yaml/
    WARNING: Retrying (Retry(total=2, connect=None, read=None, redirect=None, status=None)) after connection broken by 'SSLError("Can't connect to HTTPS URL because the SSL module is not available.")': /simple/ruamel-yaml/
    WARNING: Retrying (Retry(total=1, connect=None, read=None, redirect=None, status=None)) after connection broken by 'SSLError("Can't connect to HTTPS URL because the SSL module is not available.")': /simple/ruamel-yaml/
    WARNING: Retrying (Retry(total=0, connect=None, read=None, redirect=None, status=None)) after connection broken by 'SSLError("Can't connect to HTTPS URL because the SSL module is not available.")': /simple/ruamel-yaml/
    ERROR: Could not find a version that satisfies the requirement ruamel.yaml>=0.15 (from pre-commit-hooks) (from versions: none)
    ERROR: No matching distribution found for ruamel.yaml>=0.15
Check the log at C:\Users\Rinrada\.cache\pre-commit\pre-commit.log
```
I have looked for answers online and a few seem to be relevant but I am still not sure how to fix it, for example, [this one](https://github.com/pre-commit/pre-commit/issues/1283?fbclid=IwAR1wDO8dQ_3HfuUu-XM2NOA-6NI6o4cS_fRc5NlI6a57-bfgBSngCa6-ELI) and [this one](https://github.com/pypa/virtualenv/issues/1139?fbclid=IwAR3KX-BoxTXsDCV7p0G3M1yCRSaDAl02i85tA4tr8h6lbuSbPxJLIV5WjQw).

Any hint will be appreciated! Thank you.

**Comment by ArneBinder** [comment]
@idalr can you post step-by-step instructions for how you fixed this as a comment to this issue so that we can close it?

**Comment by idalr** [comment]
As suggested in the other Issues I cited above, this is the issue with `path` and `conda`'s environment. I solved this by installing a virtual environment via [venv](https://docs.python.org/3/library/venv.html). The package installation information is described [here](https://packaging.python.org/en/latest/guides/installing-using-pip-and-virtual-environments/#creating-a-virtual-environment).

In brief, here was what I did:
1) install `pip`
2) install `virtualenv`
3) create a virtual environment `env`
4) activate the virtual environment `env`
5) check if python is in `env` directory: `env\Scripts\python.exe`
6) install other packages using requirement files: `py -m pip install -r requirements.txt`

Following these steps, `pre-commit run -a` should be executable, given that the virtual environment is activated.
